docker/build-tools/prow-entrypoint.sh

```diff
@@ -66,64 +77,67 @@ log "Starting test..."
 sysctl net.ipv6.conf.all.forwarding=1
 sysctl net.ipv6.conf.all.disable_ipv6=0
 log "Done enabling IPv6 in Docker config."
+function run_docker() {
+  # Set ENABLE_DOCKER to what they specify, or otherwise enable if /var/lib/docker is enabled (required for docker)
+  [[ -d "/var/lib/docker" ]] && HAS_DOCKER="true"
+  ENABLE_DOCKER="${ENABLE_DOCKER:-"${HAS_DOCKER}"}"
 
-# Set ENABLE_DOCKER to what they specify, or otherwise enable if /var/lib/docker is enabled (required for docker)
-[[ -d "/var/lib/docker" ]] && HAS_DOCKER="true"
-ENABLE_DOCKER="${ENABLE_DOCKER:-"${HAS_DOCKER}"}"
-
-if [[ "${ENABLE_DOCKER}" == "true" ]]; then
-  log "Enabling docker..."
-  # Enable debug logs for docker daemon, and set the MTU to the external NIC MTU
-  # Docker will always use 1500 as the MTU; in environments where the host has <1500 as the MTU
-  # this may cause connectivity issues.
-  mkdir -p /etc/docker
-  primaryInterface="$(awk '$2 == 00000000 { print $1 }' /proc/net/route)"
-  hostMTU="$(cat "/sys/class/net/${primaryInterface}/mtu")"
-  echo "{\"debug\":true, \"mtu\":${hostMTU:-1500}}" > /etc/docker/daemon.json
-
-  # Start docker daemon and wait for dockerd to start
-  service docker start
-
-  log "Waiting for dockerd to start..."
-  while :
-  do
-    log "Checking for running docker daemon."
-    if docker ps -q > /dev/null 2>&1; then
-      log "The docker daemon is running."
-      break
-    fi
-    sleep 1
-  done
-fi
-
-function cleanup() {
   if [[ "${ENABLE_DOCKER}" == "true" ]]; then
-    log "Starting cleanup..."
-    # Cleanup all docker artifacts
-    docker ps -q | xargs -r docker kill
-    docker system prune -af || true
-    log "Cleanup complete"
+    log "Enabling docker..."
+    # Enable debug logs for docker daemon, and set the MTU to the external NIC MTU
+    # Docker will always use 1500 as the MTU; in environments where the host has <1500 as the MTU
+    # this may cause connectivity issues.
+    mkdir -p /etc/docker
+    primaryInterface="$(awk '$2 == 00000000 { print $1 }' /proc/net/route)"
+    hostMTU="$(cat "/sys/class/net/${primaryInterface}/mtu")"
+    echo "{\"debug\":true, \"mtu\":${hostMTU:-1500}}" > /etc/docker/daemon.json
+
+    # Start docker daemon and wait for dockerd to start
+    service docker start
+
+    log "Waiting for dockerd to start..."
+    while :
+    do
+      log "Checking for running docker daemon."
+      if docker ps -q > /dev/null 2>&1; then
+        log "The docker daemon is running."
+        break
+      fi
+      sleep 1
+    done
   fi
+  function cleanup() {
+    if [[ "${ENABLE_DOCKER}" == "true" ]]; then
+      log "Starting cleanup..."
+      # Cleanup all docker artifacts
+      docker ps -q | xargs -r docker kill
+      docker system prune -af || true
+      log "Cleanup complete"
+    fi
+  }
+
+  trap cleanup EXIT
+
+  # Always try to authenticate to GCR and AR.
+  gcloud auth configure-docker "${GCP_REGISTRIES}" -q || true
 }
 
-trap cleanup EXIT
+tracing::run "docker" run_docker
```

Comment:
this call to `run_docker` (and similar calls below) simply do not run as `tracing::run` does nothing, right?

Reply:
:facepalm: thanks!!

Reply:
Fixed